File: Inverted-Index/src/search/SearchEngine.java
package search;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

import file_handler.Doc;
import keywords.Keyword;

public class SearchEngine {
    private InvertedIndex index;

    public SearchEngine(InvertedIndex index) {
        this.index = index;
    }
    public Set<Doc> search(List<Keyword> keywords) {
        Set<Doc> docs = new HashSet<>();
        for (Keyword keyword : keywords) {
            
            var newDocs = index.get(keyword.getWord());
            docs = keyword.operate(docs, newDocs);
        }
        return docs;
    }
}

File: Phase5-TDD/InvertedIndex/Search/SearchEngine.cs
using System.Collections.Generic;
using System;
using InvertedIndex.Keywords;
using InvertedIndex.FileHandler;

namespace InvertedIndex.Search
{
    public class SearchEngine
    {
        private InvertedIndex index;

        public SearchEngine(InvertedIndex index)
        {
            this.index = index;
        }
        public HashSet<Doc> Search(IList<Keyword> keywords)
        {
            HashSet<Doc> docs = null;

            foreach(var keyword in keywords) {
                
                var newDocs = index.Get(keyword.GetWord());
                docs = keyword.Operate(docs, newDocs);
            }
            return docs;
        }
    }
}